packages/aws-cdk-lib/aws-synthetics/lib/canary.ts

```diff
@@ -296,6 +344,31 @@ export class Canary extends cdk.Resource implements ec2.IConnectable {
       this._connections = new ec2.Connections({});
     }
 
+    if (props.runtime.family !== RuntimeFamily.NODEJS && props.artifactS3Encryption) {
```

Comment:
Artifact encryption functionality is available only for canaries that use Synthetics runtime version syn-nodejs-puppeteer-3.3 or later. 
However, versions prior to 3.3 are deprecated and can no longer be configured, which is why I implemented it this way.

Reply:
Makes sense 👍 Can you please add a short comment about it before the check?

Reply:
Thanks. I've added a comment.